cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilder.java

```diff
@@ -283,20 +289,88 @@ private void printConfigProviders(AbstractConfiguration userConfig) {
      * Adds user provided Kafka Connect configurations.
      *
      * @param configurations   User provided Kafka Connect configurations
-     *
+     * @param injectKafkaJmxReporter         Flag to indicate if metrics are enabled. If they are we inject the JmxReporter into the configuration
+     * @param injectStrimziMetricsReporter   Inject the Strimzi Metrics Reporter into the configuration
      * @return Returns the builder instance
      */
-    public KafkaConnectConfigurationBuilder withUserConfigurations(AbstractConfiguration configurations) {
+    public KafkaConnectConfigurationBuilder withUserConfiguration(AbstractConfiguration configurations,
+                                                                  boolean injectKafkaJmxReporter,
+                                                                  boolean injectStrimziMetricsReporter) {
+        // creating a defensive copy to avoid mutating the input config
+        if (configurations instanceof KafkaConnectConfiguration kcConfiguration) {
+            configurations = new KafkaConnectConfiguration(kcConfiguration);
+        } else if (configurations instanceof KafkaMirrorMaker2Configuration kmm2Configuration) {
+            configurations = new KafkaMirrorMaker2Configuration(kmm2Configuration);
+        } else {
+            // empty user configuration
+            configurations = new KafkaConnectConfiguration(reconciliation, new ArrayList<>());
+        }
+
         printConfigProviders(configurations);
 
-        if (configurations != null && !configurations.getConfiguration().isEmpty()) {
-            printSectionHeader("Provided configurations");
+        printMetricReportersForKey("metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+        printMetricReportersForKey("admin.metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+        printMetricReportersForKey("producer.metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+        printMetricReportersForKey("consumer.metric.reporters", configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+
+        if (!configurations.getConfiguration().isEmpty()) {
+            printSectionHeader("User Provided configuration");
             writer.println(configurations.getConfiguration());
             writer.println();
         }
         return this;
     }
 
+    private void printMetricReportersForKey(String configKey,
```

Comment:
Maybe you can update this in the same way as I suggested for the Kafka configuration buider?